Automattic/wp-calypso
# Backup: add BackupRealtimeMessage component for displaying backup real-time info

Resolves https://github.com/Automattic/jetpack-backup-team/issues/615

## Proposed Changes

* Add `BackupRealtimeMessage` component for displaying backup real-time info using the new fields introduced in https://github.com/Automattic/wp-calypso/pull/93885.
* Add `BackupRealtimeMessage` unit tests
* Render `BackupRealtimeMessage` on latest backup section in the backup page
* Conditionally render `BackupRealtimeMessage` under the `jetpack/backup-realtime-message`

![CleanShot 2024-08-26 at 20 17 26@2x](https://github.com/user-attachments/assets/1b8127a9-44bf-4989-9cd8-70f8d09b1fad)

## Why are these changes being made?


For making clear daily vs real-time backups to end users.
Project thread: peaFOp-2Iy-p2

## Testing Instructions


> [!NOTE]
> It could be tricky to test all scenarios on the UI because you might need to have a site with multiple backups and real-time events, so you can rely on unit tests for this PR.
> You can run unit tests locally by running:
> 
> `yarn run test-client client/components/jetpack/daily-backup-status/test/backup-realtime-message.jsx`

* Spin up a Jetpack Cloud live branch
* Enable the feature flag appending `flags=jetpack/backup-realtime-message` to the URL
* Pick a site with Jetpack VaultPress Backup with some backups and real-time events
* Navigate to the Backup page
* Search for a date where your last backup is a real-time event. If you can't find any, you could try doing some actions on your site like publishing/updating a post or uploading images.
* If your last backup is a real-time event, you should see a message like this:
  ![CleanShot 2024-08-27 at 15 47 34@2x](https://github.com/user-attachments/assets/7892bb6e-5bfb-41ea-b02a-43f368758462)
* If it is not a real-time event, you should not see the message:
  ![CleanShot 2024-08-27 at 15 49 33@2x](https://github.com/user-attachments/assets/f7d1a427-8346-442d-81dc-ee6dd84a66c1)




## Pre-merge Checklist
